pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,879 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+"""
+Implementation of an interface class for ThermoStream® Systems devices.
+
+Authors:
+    - Markus Roeleke (markus.roeleke@bosch.com)
+
+Reference Document for implementation:
+
+ATS-515/615, ATS 525/625 & ATS 535/635 ThermoStream® Systems
+Interface & Applications Manual
+Revision E
+September, 2019
+
+Here the list of NOT implemented cmds:
+
+Command      | imp | Description
+#############+#####+#################################################################
+*TST?        | no  | Self test (dummy, always returns 0, meaning “passed”).
+-------------+-----+-----------------------------------------------------------------
+*OPC         | no  | Not implemented
+*OPC?        |     | 
+-------------+-----+-----------------------------------------------------------------
+*WAI         | no  | Not implemented
+-------------+-----+-----------------------------------------------------------------
+!            | no  | Device clear – clears the serial communications subsystem. The
+             |     | System echos back the “!” when the command has completed. If the
+             |     | “!” response is not received, the command should be retried.
+             |     | NOTE: This command is sent as a single character (no line feed
+             |     | terminator), and should never otherwise appear in a string sent to
+             |     | the System.
+-------------+-----+-----------------------------------------------------------------
+^            | no  | The System sends a “^” as a service request (SRQ) indicator in
+             |     | serial mode.
+             |     | NOTE: The “^” character never otherwise displays in a response
+             |     | string, and is sent as a single character.
+-------------+-----+-----------------------------------------------------------------
+"""
+
+from pymeasure.instruments.instrument import Instrument
+from pymeasure.instruments.validators import *
+from enum import Enum
+import time
+
+import functools
+
+GLOBAL_DELAY_FOR_CMD_EXEC = 0.05
+
+def wait_for_cmd_exec(func):
+    @functools.wraps(func)
+    def wait_after_cmd(*args, **kwargs):
+        val = func(*args, **kwargs)
+        time.sleep(GLOBAL_DELAY_FOR_CMD_EXEC)
+        return val
+    return wait_after_cmd
+
+Instrument.setting = wait_for_cmd_exec(Instrument.setting)
+Instrument.control = wait_for_cmd_exec(Instrument.control)
+Instrument.measurement = wait_for_cmd_exec(Instrument.measurement)
```

Comment:
I'm not sure if this does what you think it does. Right now, this should add 50 ms between each call when defining the Temptronic *class* -- the .setting etc.  methods are property factories that create and return a property to be attached to an Instrument subclass.
Are you maybe looking for the `query_delay` kwarg of the VISAAdapter/pyvisa?

Reply:
yes exactly, in the manual is written that there need to be a delay between cmds of 50 ms. How would you enforce this delay for all possible adapters, even future ones?